### PR TITLE
Upgrade rails to 5.2.3 with rails app:update command

### PR DESCRIPTION
**Pull request by tommotaylor**
Upgrade Rails to 5.2.3 (latest before Rails 6)
- Using `rails app:update`
- Implements bootsnap

Upgrade notes: https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-5-1-to-rails-5-2

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,7 +2,8 @@ source 'https://rubygems.org'
 
 ruby File.read(".ruby-version").strip
 
-gem 'rails', '~> 5.2.2.1'
+gem 'bootsnap', '>= 1.1.0', require: false
```
Comment (emileswarts) [inline Gemfile]: Is this the equivalent of spring?
Does Rails need this?

File: docker/api/Dockerfile [hunk 2]
```diff
@@ -10,6 +10,7 @@ WORKDIR /app
 COPY Gemfile* .ruby-version ./
 
 ARG BUNDLE_FLAGS
+RUN gem install bundler
```
Comment (emileswarts) [inline docker/api/Dockerfile]: How was this working previously?
New Rails version changed something?

File: Gemfile (hunk 1 above)
Comment (tommotaylor) [reply Gemfile]: Yeah, have a read of the upgrade notes in the description above. Don't neeeeed it but its part of Rails 5.2 and apparently helps.

File: docker/api/Dockerfile (hunk 2 above)
Comment (tommotaylor) [reply docker/api/Dockerfile]: I kept getting an error on bundler version mismatch etc etc. I think we were just stuck on the original bundler version as we weren't installing bundler each time?? Does bundler come pre-installed with ruby alpine?

Comment (tommotaylor) [reply docker/api/Dockerfile]: This has lead me down a bit of a rabbit hole as to the best way of keeping bundle up to date when using it in Dockerised apps.

**Pull request merged**